pycparser/c_lexer.py

```diff
@@ -205,10 +205,10 @@ def _make_tok_location(self, token):
     # parse all correct code, even if it means to sometimes parse incorrect
     # code.
     #
-    simple_escape = r"""([a-zA-Z._~!=&\^\-\\?'"])"""
-    decimal_escape = r"""(\d+)"""
-    hex_escape = r"""(x[0-9a-fA-F]+)"""
-    bad_escape = r"""([\\][^a-zA-Z._~^!=&\^\-\\?'"x0-7])"""
+    simple_escape = r"""([a-wyzA-Z._~!=&\^\-\\?'"]|x(?![0-9a-fA-F]))"""
```

Comment:
Please add a sizable comment explaining how these regexes are structured and why. The original regexes were taken verbatim from the C syntax definition, IIRC